src/js/electron/main.js

```diff
@@ -55,21 +54,26 @@ async function main() {
     }
   }
 
-  app.on("ready", () => {
+  function onReady() {
     installExtensions()
     if (app.commandLine.hasSwitch("new-window")) {
       winMan.openWindow("search")
     } else {
-      winMan.init(sessionState)
+      winMan.init(data)
     }
-  })
+  }
+
+  // The app might be ready by the time we get here due to async stuff above
+  if (app.isReady()) onReady()
+  else app.on("ready", onReady)
```

Comment:
why do we need more than the event listener (`app.on("ready",...`) here?

Reply:
See the comment right above it there. The async stuff above (loading and migrating the state stuff from the file) was taking longer than it took for `app` to be ready. So here we check if it's ready, then call the callback immediately, else setup the callback.